code/hdb/startup.q

```diff
@@ -1,4 +1,55 @@
 
+//called after new HDB cluster finishes loading all dependencies
+//triggers RDB reload instead of the HDB and removes old HDB cluster if it is a replacement
+.hdb.startup:{
+  gws:exec w from .servers.SERVERS where proctype = `gateway, .dotz.liveh w;
+  if[not count gws; :()]; // TODO : Handle case when none
+
+  // block queries to new hdb
+  gws @\: (`.gw.setserveractiveflag;.proc.procname;0b);
+  // wait for status change
+  .finspace.checkstatus[(.finspace.getcluster;.proc.procname);("RUNNING";"FAILURE");00:01;0D01];
```

Comment:
is this waiting for itself to be running? does this code run before the cluster is "running"? (seems confusing to me, just want to confirm I'm not misunderstanding)

Reply:
Yes, init code can run when the cluster is still in "CREATING" state